engine/syncer.go

```diff
@@ -529,9 +529,12 @@ func (e *ExchangeCurrencyPairSyncer) Start() {
 			}
 
 			if !ws.IsConnected() && !ws.IsConnecting() {
-				go WebsocketDataReceiver(ws)
+				go Bot.WebsocketDataReceiver(ws)
 
 				err = ws.Connect()
+				if err == nil {
+					err = ws.FlushChannels()
```

Comment:
Given we have an `else` after `if err != nil {` would you be able to combine the two? Either put both in `if err == nil` or in the `else`

Reply:
not quite... I'd still need to add another `err!=nil` check for after the FlushChannels and then duplicate the lines 538-543, I prefer to keep it shorter like this

Reply:
Ahh I missed that. Yeah fair enough.